reconcile/utils/sqs_gateway.py

```diff
@@ -57,12 +58,16 @@ def receive_messages(
         visibility_timeout=30,
         wait_time_seconds=20,
     ):
-        messages = self.sqs.receive_message(
-            QueueUrl=self.queue_url,
-            VisibilityTimeout=visibility_timeout,
-            WaitTimeSeconds=wait_time_seconds,
-        ).get("Messages", [])
-        return [(m["ReceiptHandle"], json.loads(m["Body"])) for m in messages]
+        try:
+            messages = self.sqs.receive_message(
+                QueueUrl=self.queue_url,
+                VisibilityTimeout=visibility_timeout,
+                WaitTimeSeconds=wait_time_seconds,
+            ).get("Messages", [])
+            return [(m["ReceiptHandle"], json.loads(m["Body"])) for m in messages]
+        except Exception as e:
+            logging.error(f"Error receiving messages from SQS: {messages=}")
+            raise e
```

Comment:
When there is exception raised in `receive_message`, `messages` will be undefined, try to access it will raise another `NameError`.

If we just want to capture `json.loads` exceptions, should only put that in try block, and just `raise` to keep stack.